src/cargo/ops/lockfile.rs

```diff
@@ -68,10 +69,29 @@ pub fn write_lockfile(dst: &Path, resolve: &Resolve) -> CargoResult<()> {
         None => {}
     }
 
+    // Try to preserve CRLF line endings, see #2076
+    if has_crlf_line_endings(dst) {
+        out = out.replace("\n", "\r\n");
+    }
+
     try!(paths::write(dst, out.as_bytes()));
     Ok(())
 }
 
+
+// It's not critical so ignore any errors.
+fn has_crlf_line_endings(path: &Path) -> bool {
+    if let Ok(file) = File::open(path) {
+        let mut buf = BufReader::new(file);
+        let mut line = String::new();
+        buf.read_line(&mut line).is_ok() && line.ends_with("\r\n")
+    }
+    else {
```

Comment:
Stylistically in Cargo the `else` is on the same line as the closing brace of the `if`